wandb/wandb_run.py

```diff
@@ -661,6 +661,36 @@ def _add_viz(self, key, viz_id):
         }
         self.config.persist()
 
+    # Stores a singleton item to wandb config.
+    #
+    # A singleton in this context is a piece of data that is continually
+    # logged with the same value in each history step, but representing
+    # as a single item in the config.
+    #
+    # We do this to avoid filling up history with a lot of repeated uneccessary data
+    #
+    # Add singleton can be called many times in one run and it will only be updated when the value changes. The last value logged will be the one persisted to the server
+    def _add_singleton(self, type, key, value):
+        # Wrap te value with information
+        value_extra = {
+            'type': type,
+            'key': key,
+            'value': value
+        }
+
+        if not key in self.config['_wandb']:
+            self.config['_wandb'][key] = {}
+
+        if type in self.config['_wandb'][key]:
+            old_value = self.config['_wandb'][key][type]
+        else:
+            old_value = None
+
+        if value_extra != old_value:
+            self.config['_wandb'][key][type] = value_extra
+
+        self.config.persist()
```

Comment:
Shouldn't this be under the check above?

A deep comparison would probably be better for the check above.